### PR TITLE
Rising bubble on simplex meshes

### PR DESCRIPTION
**Pull request by peterrum**
This PR extends the remaining ls-related classes for simplex meshes and adds a test (rising bubble).

The result look like as the following:

adaflo-simplex-bubble

and:

Screenshot from 2020-12-10 22-30-13

### CODE REVIEW

File: source/level_set_base.cc [hunk 1]
```diff
@@ -63,33 +69,71 @@ LevelSetBaseAlgorithm<dim>::LevelSetBaseAlgorithm(const FlowParameters &paramete
...
+        */
+
+        std::vector<unsigned int> lexicographic_ls(fe_mine->dofs_per_cell);
+        std::vector<unsigned int> lexicographic_p(fe_p->dofs_per_cell);
+
+        for (unsigned int j = 0; j < fe_mine->dofs_per_cell; ++j)
+          lexicographic_ls[j] = j;
+
+        for (unsigned int j = 0; j < fe_p->dofs_per_cell; ++j)
+          lexicographic_p[j] = j;
```
Comment (peterrum) [inline source/level_set_base.cc:96]: I am not sure about these lines in the context of simplex elements. I simply count up here. How are these permutations relevant?

This is related to https://github.com/dealii/dealii/pull/11353.

Comment (kronbichler) [reply source/level_set_base.cc:96]: This seems correct. They are used for the same purpose as you used the renumbering from `shape_info` in a dealii PR I reviewed today regarding the computation of a matrix from `FEEvaluation` - one needs a compatible numbering with the way `MatrixFree`/`FEEvaluation` sorts its indices. I think this function was added before we had the renumbering exposed in shape info, but one could of course now switch to shape info as well.

Comment (peterrum) [reply source/level_set_base.cc:96]: Oh I see. Thanks for the clarification! This is not relevant for simplex+MF at the moment and probably won't be relevant!?

Comment (kronbichler) [reply source/level_set_base.cc:96]: It is definitely not relevant with the implementation we have right now. However, if we were adding truncated tensor product evaluation to the simplex elements (which is in `MatrixFree` for `FE_DGP`), we would need a renumbering because `FE_P` enumerates unknowns in a hierarchical way, which is not the truncated lexicographic one.

**Review by kronbichler — APPROVED** [review]
